ui/src/App.tsx

```diff
@@ -1,11 +1,24 @@
 import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import Header from './Header';
+import Routes from './Routes';
+
+import firebase from './utils/firebase';
+
+const App: React.FC = () => {
+  const [currentUser, setCurrentUser] = React.useState<firebase.User | null>(null);
+
+  React.useEffect(() => {
+    firebase.auth().onAuthStateChanged((u) => setCurrentUser(u));
```

Comment:
Same for this. It makes handling login/logout easier, I'm not sure if having the `onAuthStateChanged` global so high in the hierarchy makes sense.